packages/expo-updates/android/src/main/java/expo/modules/updates/UpdatesController.kt

```diff
@@ -35,10 +37,23 @@ class UpdatesController {
           null
         }
 
-        singletonInstance = if (UpdatesConfiguration.canCreateValidConfiguration(context, configuration) && updatesDirectory != null) {
+        val canCreateValidConfiguration = UpdatesConfiguration.canCreateValidConfiguration(context, configuration)
+        singletonInstance = if (canCreateValidConfiguration && updatesDirectory != null) {
           val updatesConfiguration = UpdatesConfiguration(context, null)
           EnabledUpdatesController(context, updatesConfiguration, updatesDirectory)
         } else {
+          val logger = UpdatesLogger(context)
+          if (!canCreateValidConfiguration) {
+            logger.warn(
+              "The expo-updates system is disabled, either explicitly or by invalid configuration. If you wish to enable it, ensure a valid URL and runtime version are supplied.",
+              UpdatesErrorCode.InitializationError
+            )
+          } else {
```

Comment:
Is it possible to detect whether expo-updates is disabled explicitly? It would be a better developer experience to get a more precise error message like:
```
if (isDisabled) {
  `The expo-updates system is explicitly disabled. To enable it, set <something about the XML config file>`
} else (! canCreateValidConfiguration) {
  `The expo-updates system is disabled due to invalid configuration. If you wish to enable it, ensure a valid URL and runtime version are supplied.
}
```